File: leetcode/933.number-of-recent-calls.cpp
/*
 * @lc app=leetcode id=933 lang=cpp
 *
 * [933] Number of Recent Calls
 */

// @lc code=start

class RecentCounter {
public:
	queue<int>	requests;

	RecentCounter() {}

	int ping(int t)
	{
		requests.push(t);
		while (!requests.empty() && requests.front() < t - 3000) {
			requests.pop();
		}
		return (requests.size());
	}
};

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter* obj = new RecentCounter();
 * int param_1 = obj->ping(t);
 */
// @lc code=end

File: leetcode/933.number-of-recent-calls.py
#
# @lc app=leetcode id=933 lang=python3
#
# [933] Number of Recent Calls
#

# @lc code=start
class RecentCounter:

	def __init__(self):
		self.q = []

	def ping(self, t: int) -> int:
		self.q.append(t)
		while self.q and self.q[0] < t - 3000:
			self.q.pop(0)
		return len(self.q)


# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)
# @lc code=end

